Upgrade to Lucene 9.7.0.

jpountz opened the pull request:
This is the official release, and is very close to the snapshot we're currently on.

elasticsearchmachine commented:
Pinging @elastic/es-distributed (Team:Distributed)

kingherc commented on gradle/verification-metadata.xml:
```diff
@@ -2430,7 +2458,9 @@
          <artifact name="ldap-backend-1.1.1.jar">
-            <sha256 value="d3c030fe9eaeeb8a7b0844a1b48cecef20bc0a8deb5718a6c482bfc325069ee0" origin="Generated by Gradle"/>
+            <sha256 value="d3c030fe9eaeeb8a7b0844a1b48cecef20bc0a8deb5718a6c482bfc325069ee0" origin="Generated by Gradle">
+               <also-trust value="b427406dbe76279cf4fd496af17792c4c39fb6f295480290bb8be987d33cbc20"/>
```
Why do we need also-trust values?

jpountz replied:
`gradlew updateShas` did this for me, I don't know why. @breskeby Do you know about this?

kingherc reviewed (commented):
CI seems to be failing

jpountz commented:
> CI seems to be failing

Indeed, not all artifacts were available on Maven Central yet based on the output, I'll rerun tests.

@elasticsearchmachine test this please

jpountz commented:
It looks like it's more complicated than that, I'll dig.

jpountz committed:
Upgrade to Lucene 9.7.0.

This is the official release, and is very close to the snapshot we're currently on.

jpountz committed:
Remove side effects to verification metadata that are unrelated to Lucene.

jpountz commented:
Re-computing verification metadata did the trick, it's green now.

ChrisHegarty reviewed (approved):
LGTM.

iverase commented on gradle/verification-metadata.xml:
```diff
@@ -2488,121 +2488,241 @@
+      <component group="org.apache.lucene" name="lucene-analysis-common" version="9.7.0">
+         <artifact name="lucene-analysis-common-9.7.0.jar">
+            <sha256 value="87abf0221c526d64616d47d2261866620c8f42659c93dbfd0d1ffb138af98137" origin="Generated by Gradle"/>
+         </artifact>
```
We should remove the old entries, unfortunately the command don't do it for us and we should prevent the file to become huge

iverase replied:
than you so much Adrien!

jpountz committed:
Remove verification metadata entries from previous Lucene snapshot.

iverase reviewed (approved):
LGTM

iverase commented:
@jpountz in case you miss it, it seems backport bot didn't work this time.

jpountz commented:
I did miss it, thanks @iverase!
